### PR TITLE
AzureMonitor: Correctly set filter values in portal URL

### LINKED ISSUE #48455: AzureMonitor: View in Azure Portal link does not correctly display for queries with dimension filters
**What happened**:

Attempting to view the relevant chart in the Azure portal by selecting `View in Azure Portal` on a data point where the query includes a dimension filter leads to a view that does not display the filter.

image

image

**What you expected to happen**:

We'd expect the link to correctly navigate to a chart that displays the filter and filtered data.

image

### PR DESCRIPTION
**Pull request by aangelisc**
Correctly set Azure portal URL for queries containing dimensions.

Fixes #48455

### CODE REVIEW

**Review by andresmgot — COMMENTED** [review]
you beat me to this :P (but you forgot the tests :joy:)

File: pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go [hunk 1]
```diff
@@ -361,7 +361,15 @@ func (e *AzureMonitorDatasource) parseResponse(amr types.AzureMonitorResponse, q
...
 		}
 
-		queryUrl, err := getQueryUrl(query, azurePortalUrl, resourceID, resourceName)
+		queryJSONModel := types.AzureMonitorJSONQuery{}
+		err := json.Unmarshal(originalQuery.JSON, &queryJSONModel)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode the Azure Monitor query object from JSON: %w", err)
+		}
+
+		dimensions := MigrateDimensionFilters(queryJSONModel.AzureMonitor.DimensionFilters)
```
[the remark this answers is not in the document]
Comment (aangelisc) [reply pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go]: Great shout!

File: pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go [hunk 2]
```diff
@@ -374,7 +382,7 @@ func (e *AzureMonitorDatasource) parseResponse(amr types.AzureMonitorResponse, q
 }
 
 // Gets the deep link for the given query
-func getQueryUrl(query *types.AzureMonitorQuery, azurePortalUrl, resourceID, resourceName string) (string, error) {
+func getQueryUrl(query *types.AzureMonitorQuery, azurePortalUrl, resourceID, resourceName string, dimensions []types.AzureMonitorDimensionFilter) (string, error) {
```
[the remark this answers is not in the document]
Comment (aangelisc) [reply pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go]: I've updated the existing build query tests which should cover the new logic, let me know if this is okay 👍

**Review by andresmgot — APPROVED** [review]
LGTM! Good job

**Pull request merged**